shared/core/src/firebase/observable-db-ref.ts

```diff
@@ -59,6 +57,9 @@ export class ObservableDbRef {
   }
 
   on(eventName: string): Observable<firebase.database.DataSnapshot> {
-    return fromEventPattern(handler => this.ref.on(eventName, handler), handler => this.ref.off(eventName, handler));
+    return fromEventPattern(
+      handler => this.ref.on(eventName, (val: firebase.database.DataSnapshot) => handler(val)),
```

Comment:
We needed to change this to a lambda, otherwise this event emits an `Array<DataSnapshot>` instead of `DataSnapshot`. Could nail down why this is, but it happened after upgrading rxjs

Reply:
In addition we have to add the `DataSnapshot` type as it's not being inferred correctly otherwise.